src/pool/srv_pool.c

```diff
@@ -3775,6 +3775,257 @@ pool_find_all_targets_by_addr(uuid_t pool_uuid,
 	return rc;
 }
 
+static int
+redist_open_hdls_send_rpcs(uuid_t pool_uuid, d_iov_t *handles,
+			   d_rank_list_t *ranks)
+{
+	crt_rpc_t			*tf_req;
+	struct pool_tgt_fetch_hdls_in	*tf_in;
+	struct pool_tgt_fetch_hdls_out	*tf_out;
+	crt_opcode_t			opc;
+	int				rc = DER_SUCCESS;
+	int				i;
+
+	/* Send an RPC to each rank with all of the open handles */
+	for (i = 0; i < ranks->rl_nr; i++) {
+		crt_endpoint_t svr_ep;
+
+		svr_ep.ep_grp = NULL;
+		svr_ep.ep_rank = ranks->rl_ranks[i];
+		svr_ep.ep_tag = daos_rpc_tag(DAOS_REQ_POOL, 0);
+		opc = DAOS_RPC_OPCODE(POOL_TGT_FETCH_HDLS, DAOS_POOL_MODULE, 1);
+		rc = crt_req_create(dss_get_module_info()->dmi_ctx, &svr_ep,
```

Comment:
Is it a temporary workaround? Otherwise, I think we need to consider bulk transfer when the buffer size is too large.

Reply:
Hmmm... I hadn't really considered what would happen if there are too many handles. I'll add a TODO comment to document this. Hopefully that won't be a problem when this code gets migrated to IV distribution.

Reply:
Added JIRA ticket DAOS-5232 to track this.

Reply:
useful to add log message here